editor/modes/visual-editor/block.js

```diff
@@ -321,7 +321,7 @@ class VisualEditorBlock extends Component {
 
 		// Generate the wrapper class names handling the different states of the block.
 		const { isHovered, isSelected, isMultiSelected, isFirstMultiSelected, focus } = this.props;
-		const showUI = isSelected && ( ! this.props.isTyping || focus.collapsed === false );
+		const showUI = isSelected && ( ! this.props.isTyping || ( focus && focus.collapsed === false ) );
```

Comment:
Minor / unrelated: Do we need to check that this is explicitly false, or could we simplify to `! focus.collapsed` ?